### PR TITLE
SitePreview: Update externalUrl to represent the iframes location

### PR DESCRIPTION
**Pull request by spen**
### Summary
These changes add the ability for the SitePreview to listen for location changes to the site being previewed and update a newly added URL bar and existing external link button accordingly.

screen shot 2017-07-18 at 16 03 39

To Do:
- Possibly update the way in which the URL is displayed? Should we be showing the <kbd>Copy</kbd> button?

### Test Plan
- Go to `calypso.localhost:3000/view/youramazingsite.wordpress.com` to preview your site.
- Navigate around your site and each time you click a link:
    - check first that the external link has a href that represents the current page: screen shot 2017-07-11 at 01 41 12
    - check that the URL bar above the site preview also represents the current page

### Extra Testing
During reviews some points were raised, here are the steps to test that each issue has been squished (*In progress*):
- [x] Unmapped domain showing instead of mapped, primary domain:
    - Test with a site that has a mapped domain set to be primary
    - Ensure that the URL shows this domain and not the original `xx.wordpress.com` domain
    - Bonus points: switch the primary domain to another domain (can be `xx.wordpress.com`) and ensure that this domain now shows in the URL bar
- [x] Trailing slash appears after the iframe page loads, but is not shown while loading
    - Watch the URL bar when initially landing on`/view/your-site` note that when on your homepage there is no trailing `/`.
    - Navigate to a page on your site, there should also be no trailing `/` at the end of the path (i.e. `your-site.com/about` and not `your-site.com/about/`
- [x] trim the `scheme://` from the URL, to save some space and better display longer URLs.
    - Make sure that `https://` or `http://` is not shown in the bar.

### CODE REVIEW

**Comment by marekhrabe** [comment]
This PR can be tested by sandboxing `yoursite.wordpress.com` and patching your sandbox with D6323-code

**Comment by marekhrabe** [comment]
I noticed a tiny little thing - when you initially load `/view/site` it shows the domain without the ending slash (example: `https://theplate.blog`) and after the page loads, the ending `/` is added. Can we make `/` a default pathname so it wouldn't jump around? :)

**Comment by spen** [comment]
> This PR can be tested by sandboxing yoursite.wordpress.com and patching your sandbox with D6323-code

Changes to WPCOM are now made - this should now be testable without the need to sandbox :)

**Comment by alisterscott** [comment]
Looking good, a couple of notes:

1. My site has primary domain `watirmelon.blog`, but the address is showing as `watirmelon.wordpress.com`
2. On wide monitors, the URL area is still narrower than space available - which means I can't see the whole URL for long permalinks, is it possible to expand it when the space is available? (see image below)

screen shot 2017-07-20 at 11 51 52 am

**Comment by spen** [comment]
Thanks for the feedback @alisterscott & @marekhrabe :)
I'll work on those points today :D

**Comment by marekhrabe** [comment]
The code that is building the `externalUrl` should still use `site.URL` instead of the `getBasePreviewUrl` function to get the prefix.

**Comment by marekhrabe** [comment]
From todo list of this PR:
> Navigate to a page on your site, there should also be no trailing / at the end of the path (i.e. your-site.com/about and not your-site.com/about/

I think we should only remove the trailing slash when you are at the homepage of site — that's exactly copying Chrome behavior. Removing the trailing slash for other pages will cause generating a non-canonical link that we will always have to redirect (try for example `curl --head https://automattic.com/about` to see how we redirect users to slash version).

I also noticed that you are testing for the presence of initial slash in the pathname. That should not be necessary, as per specs it always includes the initial slash: https://developer.mozilla.org/en-US/docs/Web/API/HTMLHyperlinkElementUtils/pathname

I think with all this, we might simplify the URL generator to something like `this.props.site.URL + ( pathname === '/' ? '' : pathname )`

**Comment by spen** [comment]
Awesome :D I did think that seemed overkill but figured I'd cover against WPCOM implementation changes... Still it's highly unlikely that that would happen and we could just make a fix at that point!

**Comment by spen** [comment]
@alisterscott it looks like there are a bunch of intricacies to sizing the component relative to it's content/value, which would be necessary for the style improvements you mentioned.
I'm pretty determined to make those improvements but it might be wise to iterate with those.
For now I'm going to look at chopping the `scheme://` from the URL, which will be a small improvement to saving space at least.

**Comment by marekhrabe** [comment]
Also, by hiding the scheme, you'll prevent second bug from showing up - which is a good thing :D We currently don't track http/https very well and even for mapped dotcom sites (where we are 100% sure they support SSL), we still have just "http" in `site.URL`. We'll get to this later so it's awesome we won't bring any extra attention to it now :)

**Comment by designsimply** [comment]
<blockquote>1. My site has primary domain `watirmelon.blog`, but the address is showing as `watirmelon.wordpress.com`</blockquote>

Also reported at #16470.

**Comment by spen** [comment]
Thanks Sheri! I'll make sure that's ironed out once we finalise & merge here :)

File: client/components/clipboard-button-input/index.jsx [hunk 1]
```diff
@@ -51,13 +53,14 @@ export default React.createClass( {
...
-		const { value, className, disabled } = this.props;
+		const { value, className, disabled, hideHttp } = this.props;
 		const classes = classnames( 'clipboard-button-input', className );
 
 		return (
 			<span className={ classes }>
 				<FormTextInput
-					{ ...omit( this.props, 'className' ) }
+					{ ...omit( this.props, 'className', 'hideHttp' ) }
+					value={ hideHttp ? withoutHttp( value ) : value }
```
Comment (spen) [inline client/components/clipboard-button-input/index.jsx]: I'm having second thoughts about this tweak to the ClipboardButtonInput component - it works for this case but feels like more of a shoe-horned solution than a proper, reasonable one.

I'd love to hear opinions on it but I'm putting the PR on hold for the time being.

There was discussion of changing how the URL bar displays - some of the ideas being considered would remove the need to hide the scheme in the ClipboardInput, instead showing the URL sans-scheme in plain text. We'd be making the change at implementation level, rather making more general changes to a component needlessly.

One such solution might look a little like this:
clipboardhover

**Comment by drw158** [comment]
The url input is centered, but the text is left-aligned so it looks off centered. We definitely should avoid truncating with so much white space available. The url text should be $gray-text-min for accessibility.

Overall, the copy button really doesn't do much here. I'd keep it simple text. The user can click the external icon to go to the site.

**Comment by spen** [comment]
Awesome :D Thanks Marek!

A heads up to @alisterscott & @drw158, I've made some style improvements in PR #16524 and updated the post-post preview to show the primary domain to keep consistent with the changes here @ #16512 - Thanks again for the feedback!

**Pull request merged**